varats/experiments/wllvm.py

```diff
@@ -72,10 +82,37 @@ class Extract(actions.Step):  # type: ignore
     DESCRIPTION = "Extract bitcode out of the execution file."
 
     BC_CACHE_FOLDER_TEMPLATE = "{cache_dir}/{project_name}/"
-    BC_FILE_TEMPLATE = "{project_name}-{binary_name}-{project_version}.bc"
 
-    def __init__(self, project: Project) -> None:
+    @staticmethod
+    def get_bc_file_name(
+        project_name: str, binary_name: str, project_version: str,
+        bc_file_extensions: tp.List[BCFileExtensions]
+    ):
+        """Parses parameter information into a filename template to name a
+        bitcode file."""
+
+        if bc_file_extensions:
+            project_bc_file_ext = '-'
+
+            for ext in bc_file_extensions[:-1]:
+                project_bc_file_ext += (ext.value + '_')
+
+            project_bc_file_ext += bc_file_extensions[-1].value
```

Comment:
```suggestion
            ext_sep = ""
            for ext in bc_file_extensions:
                project_bc_file_ext += (ext_sep + ext.value)
                ext_sep =  '_'
```
That looks kinda hacky with indexing and list shorting, with this code pattern where we use a prefix it is easier to have separators between names without indexing.